FireboltPrivacy/FireboltPrivacyImplementation.cpp

```diff
@@ -0,0 +1,138 @@
+                : Core::JSON::Container()
+                , devicemanifest("")
+                , fireboltconfigpath("")
+            {
+                Add(_T("devicemanifest"), &devicemanifest);
+                Add(_T("fireboltconfigpath"), &fireboltconfigpath);
+            }
+            ~Config() override = default;
+        public:
+            Core::JSON::String devicemanifest;
+            // This path is specific to firebolt. It points to the path where
+            // firebolt config files are placed. These files are shared between Ripple and other FireboltThunderPlugins
+            // In prod it points to /etc/
+            // and in dev it points to /opt/
+            Core::JSON::String fireboltconfigpath;
+        };
+
+
+    public:
+        FireboltPrivacyImplementation(const FireboltPrivacyImplementation&) = delete;
+        FireboltPrivacyImplementation& operator= (const FireboltPrivacyImplementation&) = delete;
+        FireboltPrivacyImplementation()
+            : _adminLock()
+            , _config()
+            , _deviceManifest()
+            , _allowResumePoints(false)
+        {
+        }
+        ~FireboltPrivacyImplementation() override
+        {
+            _deviceManifest.clear();
+            if (_service) {
+                _service->Release();
+                _service = nullptr;
+            }
+        }
+
+        uint32_t Register(Exchange::IFireboltPrivacy::INotification* notification) override
+        {
+            ASSERT(notification);
+            _adminLock.Lock();
+            notification->AddRef();
+            _notifications.push_back(notification);
+            _adminLock.Unlock();
+
+            return Core::ERROR_NONE;
+        }
+
+        uint32_t Unregister(Exchange::IFireboltPrivacy::INotification* notification) override
+        {
+            ASSERT(notification);
+            _adminLock.Lock();
+            auto item = std::find(_notifications.begin(), _notifications.end(), notification);
+            ASSERT(item != _notifications.end());
+            _notifications.erase(item);
+            (*item)->Release();
+            _adminLock.Unlock();
+
+            return Core::ERROR_NONE;
+        }
+        uint32_t AllowResumePoints(bool& allowResumePoints /* @out */) const override {
+            _adminLock.Lock();
+            allowResumePoints = _allowResumePoints;
+            _adminLock.Unlock();
+	    return 0;
+        }
+        uint32_t SetAllowResumePoints(const bool& allowResumePoints ) override {
+            _adminLock.Lock();
+            _allowResumePoints = allowResumePoints;
+	    for(auto notification: _notifications){
+	        notification->OnAllowResumePointsChanged(allowResumePoints);
+	    }
+            _adminLock.Unlock();
+	    return 0;
+        }
+        uint32_t Configure(PluginHost::IShell* service)  override {
+	    ASSERT( service != nullptr);
+            SYSLOG(Logging::Notification, (_T("service Config: %s"), service->ConfigLine().c_str()));
```

Comment:
Why always log the content of the config file??? This will flood the logging with useless info. Syslog is for info valuable for operators, traces for devolopers. The operator cannot act upon this info, but I also think a trace is useless, that makes more sense after 117 to print the actual value of deviceManifest.